plugins/woocommerce-admin/client/tasks/fills/tax/index.tsx

```diff
@@ -92,20 +92,34 @@ const Tax: React.FC< TaxProps > = ( { onComplete, query, task } ) => {
 		}
 	}, [] );
 
-	const onAutomate = useCallback( () => {
+	const onAutomate = useCallback( async () => {
 		setIsPending( true );
-		updateAndPersistSettingsForGroup( 'tax', {
-			tax: {
-				...taxSettings,
-				wc_connect_taxes_enabled: 'yes',
-			},
-		} );
-		updateAndPersistSettingsForGroup( 'general', {
-			general: {
-				...generalSettings,
-				woocommerce_calc_taxes: 'yes',
-			},
-		} );
+		try {
+			await Promise.all( [
+				updateAndPersistSettingsForGroup( 'tax', {
+					tax: {
```

Comment:
Good point! I took a quick look, and I think it's okay for new users because there are only default tax settings. But we should better add it to the dependency array to avoid some edge cases. I'll create an issue for it.

Reply:
issue created https://github.com/woocommerce/woocommerce/issues/36502